llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -1388,6 +1386,7 @@ bool RISCVVLOptimizer::runOnMachineFunction(MachineFunction &MF) {
   while (!Worklist.empty()) {
     assert(MadeChange);
     MachineInstr &MI = *Worklist.pop_back_val();
+    assert(isCandidate(MI));
```

Comment:
Is this assert redundant? isCandidate explicitly blocks insertion to the Worklist in `PushOperands`. That being said, isCandidate is cheap, so I don't have a strong objection.

Reply:
We can keep it. It is cheap.